internal/middleware/account_test.go

```diff
@@ -11,26 +11,50 @@ import (
 	"github.com/RHEnVision/provisioning-backend/internal/dao/stubs"
 	"github.com/RHEnVision/provisioning-backend/internal/middleware"
 	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/aws/smithy-go/ptr"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestAccountMiddleware(t *testing.T) {
-	ctx := identity.WithIdentity(t, context.Background())
-	ctx = stubs.WithAccountDaoOne(ctx)
+	t.Run("existing account", func(t *testing.T) {
```

Comment:
so I am bit out of good docs on it, but basically it is a grouping mechanism, it is quite simmilar to something like `TestAccountMiddlewareWithExistingAccount` +  `TestAccountMiddlewareWithNewAccount`, but this subgrouping is quite useful IMHO, it helps if you want to share setup for example, but mainly makes it bit more readable, one Test method for single go method, with multiple sub-test by `Run` ... if we keep this approach I think it will be easier to lookup the appropriate tests in the future.

Reply:
So each Run method is a separate test? 
 Can we share the setup between a few Runs? 
 that is possible when the setup is out of the function scope?  (function with` func(t *testing.T)` signature)

Reply:
it is certainly possible, but need to be approached with care, because only very static data should be shared. If you share dynamic data, tests could start interfering one with another, and that's quite bad ^_^
So they can share data, but should do with care. Grouping is much better usecase for this.